Makefile

```diff
@@ -0,0 +1,53 @@
+WORKDIR := /tsfresh
+TEST_IMAGE := tsfresh-test-image
+TEST_DOCKERFILE := Dockerfile.testing
+TEST_CONTAINER := tsfresh-test-container
+# >= 3.9.2 ---> https://github.com/dask/distributed/issues/7956
+PYTHON_VERSIONS := "3.7.12 3.8.12 3.9.12 3.10.12 3.11.0"
+BLACK_VERSION := 22.12.0
+# Isort 5.12.0 not supported with python 3.7.12
+ISORT_VERSION := 5.12.0
+
+build-testenv:
+	docker build \
+			-f $(TEST_DOCKERFILE) \
+			-t $(TEST_IMAGE) \
+			--build-arg PYTHON_VERSIONS=$(PYTHON_VERSIONS) \
+			.
+
+# Tests `PYTHON_VERSIONS`, provided they are also
+# specified in setup.cfg `envlist`
+test-all-testenv: clean build-testenv
+	docker run --rm \
+			--name $(TEST_CONTAINER) \
+			-v .:$(WORKDIR) \
+			-v build_artifacts:$(WORKDIR)/build \
+			-v tox_artifacts:$(WORKDIR)/.tox \
```

Comment:
FYI build_artifacts and tox_artifacts are a 'hack'. If no docker volume is specified then they end up in the context of the .:$(WORKDIR) mapping. The issue there is that when .tox and build directories are created in the docker container, they are created by the root user but end up in the source code. This means that have root permissions not user permissions. 


Explicitly using docker volumes build_artifacts and tox_artifacts gets around this as they are removed after the container finishes running.